### PR TITLE
Improve performance of the generator

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Filters Swagger types up front by excluding resources from groups that we're not interested in exporting. This reduces the number of generated types from 120,589 to 61,904 (reduced by 49%) and improves performance both by skipping parsing of half the object model and by reducing the amount of work done by later pipeline stages.

I expect the lower memory load will also contribute to faster CI runs, given those run on constrained machines.

Also included
* Abort pipeline if the context is cancelled, allowing the generator to be interrupted 
* Some paths were being resolved from the current directory, not from the folder containing the configuration file

**How does this PR make you feel**:
gif

### CODE REVIEW

File: v2/tools/generator/internal/codegen/debug_reporter.go [hunk 1]
```diff
@@ -6,12 +6,13 @@
...
 
 import (
-	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/reporting"
-	"path"
+	"path/filepath"
 	"regexp"
 	"strconv"
 	"strings"
 
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/reporting"
```
Comment (matthchr) [inline v2/tools/generator/internal/codegen/debug_reporter.go:14]: minor: I think this should be at the bottom with the other aso v2 tools imports

File: v2/tools/generator/internal/config/configuration.go [hunk 2]
```diff
@@ -282,6 +294,11 @@ func (config *Configuration) initialize(configPath string) error {
 		errs = append(errs, errors.Errorf("destination Go module must be specified"))
 	}
 
+	// Ensure config.DestinationGoModuleFile is a fully qualified path
+	if !filepath.IsAbs(config.DestinationGoModuleFile) {
```
Comment (matthchr) [inline v2/tools/generator/internal/config/configuration.go:298]: Curious: what was the issue was with this being a relative path?

File: v2/tools/generator/internal/config/object_model_configuration.go [hunk 3]
```diff
@@ -37,6 +37,27 @@ func NewObjectModelConfiguration() *ObjectModelConfiguration {
 	}
 }
 
+func (omc *ObjectModelConfiguration) IsGroupExpected(pkg astmodel.PackageReference) bool {
```
Comment (matthchr) [inline v2/tools/generator/internal/config/object_model_configuration.go:40]: Would it make more sense to call this `ContainsGroup` or `HasGroup` or something?

`IsGroupExpected` reads weird to me. Expected for what?

File: v2/tools/generator/internal/config/configuration.go [hunk 4]
```diff
@@ -349,6 +366,15 @@ func (config *Configuration) ShouldPrune(typeName astmodel.TypeName) (result Sho
 		}
 	}
 
+	// If the type comes from a group that we don't expect, prune it.
+	// We don't also check for whether the version is expected because it's common for types to be shared
+	// between versions of an API. While end up pulling them into the package alongside the resource, at this
+	// point we haven't done that yet, so it's premature to filter by version.
+
+	if !config.ObjectModelConfiguration.IsGroupExpected(typeName.PackageReference) {
+		return Prune, fmt.Sprintf("No resources expected from %s", typeName.PackageReference.PackagePath())
```
Comment (matthchr) [inline v2/tools/generator/internal/config/configuration.go:375]: consider rephrasing to: "No resources configured for export from: %s"?

**Review by matthchr — APPROVED** [review]
LGTM - few minor comments

**Review by super-harsh — APPROVED** [review]

**Pull request merged**